### PR TITLE
Add initial modules

### PR DESCRIPTION
**Pull request by mattmillr**
* Adds decorator decorator and tests
 * Adds JSON Encoder
 * Adds linter
 * Update readme & setup.py

### CODE REVIEW

**Comment by mattmillr** [comment]
@rcatlin for review

File: chassis/tools/lint.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    FILES.extend(
+        os.path.join(dirpath, filename)
+        for filename in filenames
+        if ".py" == filename[-3:]
+    )
+
+
+# A list of messages that should not be printed by pylint.
+SUPRESSED_MESSAGES = [
+    # 'I0011',  # Inline option disables a message or a messages category.
```
Comment (rcatlin) [inline chassis/tools/lint.py]: Let's remove these supressed messages if we're not supressing them.

Comment (rcatlin) [inline chassis/tools/lint.py]: Due to our discussion, let's leave these in for now.

**Comment by rcatlin** [comment]
:+1:

**Pull request merged**